.drone.yml

```diff
@@ -903,26 +910,45 @@ steps:
       # Enterprise FIPS
       - docker build --target teleport-fips --build-arg DOWNLOAD_TYPE=teleport-ent --build-arg EXTRA_DOWNLOAD_ARGS="-fips" --build-arg VERSION_TAG=$VERSION_TAG --build-arg OS=$OS --build-arg ARCH=$ARCH -t $ENT_FIPS_IMAGE_NAME -f /go/build/Dockerfile-cron /go/build
       - docker push $ENT_FIPS_IMAGE_NAME
+      # Login to ECR
+      - docker logout quay.io
+      - aws ecr-public get-login-password --region=us-east-1 | docker login -u="AWS" --password-stdin public.ecr.aws
+      # OSS ECR
+      - docker tag $OSS_IMAGE_NAME $OSS_IMAGE_NAME_ECR
+      - docker push $OSS_IMAGE_NAME_ECR
```

Comment:
Do/should these get staged?  Or do they go directly to production?

Reply:
They currently don't get staged they are pushed directly to production.They couldn't be staged with the current Immutable rules of the new staging repository. Irregardless of the immutable rules, I'm trying to imaging the workflow for staging these, since they are not triggered by an engineer. Would we just automatically promote them to production? 

Potentially to test if the latest floating version matched the latest release then it would promote it from staging to production?

Reply:
> I'm trying to imaging the workflow for staging these, since they are not triggered by an engineer. Would we just automatically promote them to production?

I'm ok skipping staging for these if we need to do so for expediency

It does leave me a touch uncomfortable, as we're establishing a pattern from the day 0 where production is not a subset of staging.  Accounting wise, it'd be nice to trace every artifact through staging first such that we have an "audit log".

If I were to spit ball a solution, I'd publish these to staging with a date suffix `v10-20220612` and then immediately promote that version to the floating prod tag `v10`.  This leaves us with a full and immutable trail of every date based image we ever published.  If we suspect one was misbehaving or every want to rewind, we have that data.

Reply:
> It does leave me a touch uncomfortable, as we're establishing a pattern from the day 0 where production is not a subset of staging.

This pattern has existed within the teleport repository for 2 years now. I kept it this way to stick to the current status quo, not for expediency. 

> If I were to spit ball a solution, I'd publish these to staging with a date suffix v10-20220612 and then immediately promote that version to the floating prod tag v10. This leaves us with a full and immutable trail of every date based image we ever published. If we suspect one was misbehaving or every want to rewind, we have that data.

👍🏻 This doesn't seem difficult if we are fine with immediately promoting.

Reply:
> > It does leave me a touch uncomfortable, as we're establishing a pattern from the day 0 where production is not a subset of staging.
> 
> This pattern has existed within the teleport repository for 2 years now. I kept it this way to stick to the current status quo, not for expediency.

Fair point.

Reply:
I had a go at this in https://github.com/gravitational/teleport/pull/14347/commits/c5a32528f3503a6adc4430f18055ebaedb88c93e. I still need to test these changes and will be doing so soon.